ui/src/app/components/buttons.tsx

```diff
@@ -283,6 +284,38 @@ export const Button = ({
   );
 };
 
+export const ButtonWithLocationState = ({
+  children,
+  path = '',
+  type = 'primary',
+  style = {},
+  disabled = false,
+  propagateDataTestId = false,
+  ...props
+}) => {
+  // `fp.omit` used to prevent propagation of test IDs to the rendered child component.
+  const childProps = propagateDataTestId
+    ? props
+    : fp.omit(['data-test-id'], props);
+  const computedStyle = fp.merge(
+    computeStyle(buttonVariants[type], { disabled }),
+    { style }
+  );
+  return (
+    <Link
+      to={{
+        pathname: path,
+        state: NAVIGATING_FROM_WORKBENCH + location.pathname,
```

Comment:
Is there advantage to using useLocation instead?

Reply:
So i was thinking since from history i am taking the location object, so why not use it directly